matrix/Quaternion.hpp

```diff
@@ -186,6 +186,47 @@ class Quaternion : public Vector<Type, 4>
         }
     }
 
+    /**
+     * Quaternion from two vectors
+     * Generates shortest rotation from source to destination vector
+     * Default destination if not specified is [0,0,1]
+     *
+     * @param dst destination vector (no need to normalize)
+     * @param src source vector (no need to normalize)
+     * @param eps epsilon threshold which decides if a value is considered zero
+     */
+    Quaternion(const Vector3<Type> &src, const Vector<Type, 3> dst = Vector3<Type>(0, 0, 1), const Type eps = 1e-5f) :
```

Comment:
- Vector3<type> dst = Vector3<Type>(0,0,1)
- const Type eps = (Type)1e-5;

Reply:
There was `Vector<Type, 3>` which I switched to `Vector3<Type>` now, that's what you meant right?
`const Type eps = (Type)1e-5` is enough, constants are casted auatomatically to the correct type on compile time.

Reply:
You're right. Second statement is wrong... I put the explicit conversion now.